ironfish/src/wallet/wallet.ts

```diff
@@ -698,38 +659,47 @@ export class Wallet {
     transactionExpirationDelta: number,
     expiration?: number,
   ): Promise<Transaction> {
-    const heaviestHead = this.chain.head
-    if (heaviestHead === null) {
-      throw new Error('You must have a genesis block to create a transaction')
-    }
-
-    expiration = expiration ?? heaviestHead.sequence + transactionExpirationDelta
-    if (isExpiredSequence(expiration, this.chain.head.sequence)) {
-      throw new Error('Invalid expiration sequence for transaction')
-    }
-
     const raw = await this.createTransaction(
       account,
       [],
       [],
       [{ assetId, value }],
       fee,
+      transactionExpirationDelta,
       expiration,
     )
 
-    const transaction = await this.postTransaction(raw)
+    const transaction = await this.postTransaction(raw, memPool)
 
-    const verify = this.chain.verifier.verifyCreatedTransaction(transaction)
-    if (!verify.valid) {
-      throw new Error(`Invalid transaction, reason: ${String(verify.reason)}`)
-    }
+    return transaction
+  }
 
-    await this.addPendingTransaction(transaction)
-    memPool.acceptTransaction(transaction)
-    this.broadcastTransaction(transaction)
-    this.onTransactionCreated.emit(transaction)
+  async createTransactionWithFeePriorityLevel(
+    sender: Account,
+    receives: {
+      publicAddress: string
+      amount: bigint
+      memo: string
+      assetId: Buffer
+    }[],
+    mints: MintDescription[],
+    burns: BurnDescription[],
+    priorityLevel: PriorityLevel,
+    memPool: MemPool,
+    transactionExpirationDelta: number,
+    expiration?: number | null,
+  ): Promise<RawTransaction>{
+    const estimatedFee = await memPool.feeEstimator.estimateFee(priorityLevel, sender, receives)
```

Comment:
`estimateFee` will do most of the same work that `createTransaction` is doing (iterating over notes, creating witnesses).

Will we move the logic for getting a fee based on a fee rate and a raw transaction in a separate PR?

Reply:
Another way we could do this would be to modify the `fund` method to optionally take a fee rate.

`fund` takes a raw transaction and adds the spends needed to satisfy the transaction receives and fee. If it takes a fee rate it would need to do this in steps:
1. fund the receives
2. calculate the fee based on the size of the raw transaction
3. fund the fee